### PR TITLE
feat: Add light version to all input types

### PR DESCRIPTION
**Pull request by alisonjoseph**
Ref https://github.ibm.com/carbon/issues/issues/736

Add light prop to the following components

- Date Picker
- Number Input
- Search
- Select
- Textarea
- Text Input
- Time Picker
- Data Table Search (updated to light)

### CODE REVIEW

File: .storybook/_container.scss [hunk 1]
```diff
@@ -1,4 +1,4 @@
 $css--font-face: true;
 $css--reset: true;
 
-@import '~carbon-components/scss/globals/scss/styles.scss';
+@import 'node_modules/carbon-components/src/globals/scss/styles';
```
Comment (joshblack) [inline .storybook/_container.scss]: Is this something that we'll need to change after PR reviews or is this a permanent change?

Comment (alisonjoseph) [reply .storybook/_container.scss]: Oh, good catch, I'll swap it back before we merge.

File: src/components/Select/Select.js [hunk 2]
```diff
@@ -47,6 +44,11 @@ const Select = ({
           className="bx--select__arrow"
           description={iconDescription}
         />
+        {!inline ? (
```
Comment (asudoh) [inline src/components/Select/Select.js]: Is moving the `<label>` here intentional?

Comment (alisonjoseph) [reply src/components/Select/Select.js]: Yes, the html was updated in the vanilla repo, so this is just matching the changes to the v9 component there.

**Pull request merged**